src/common/options.cc

```diff
@@ -5691,7 +5691,7 @@ std::vector<Option> get_rgw_options() {
     .set_description("The maximum number of metadata items that can be put via single request"),
 
     Option("rgw_override_bucket_index_max_shards", Option::TYPE_UINT, Option::LEVEL_DEV)
-    .set_default(0)
+    .set_default(64)
```

Comment:
this config variable is an override for `RGWZone::bucket_index_max_shards`, so if we're changing the default value it should be done in the zone instead:
```diff
-  RGWZone() : log_meta(false), log_data(false), read_only(false), bucket_index_max_shards(0),
+  RGWZone() : log_meta(false), log_data(false), read_only(false), bucket_index_max_shards(8),
```
this change would only take effect on zone creation, so the release notes should mention that existing `bucket_index_max_shards` configuration can be changed in json using `radosgw-admin zonegroup get/set`

Reply:
@cbodley Let's see how things test out once #30853 is ready to go.  I suspect we're going to still see significant write-throughput gains with more per-bucket parallelism at low object counts and the listing throughput is still quite high at 64 shards.  Do you have other concerns with higher initial bucket counts?

Reply:
despite the wins we're seeing from https://github.com/ceph/ceph/pull/30853, listing is still going to be cheaper/faster with fewer shards

it is clear that a large shard count optimizes for throughput on a single-bucket workload. but i feel that workloads that require lots of parallism will either be spread across several buckets, or operating on larger buckets. i think the ~100k objects-per-shard is a fairly good fit for this in general, so my goal here is just to get _some_ parallelism by default. and i think an initial count of ~8 strikes a good balance